core/divesitehelpers.cpp

```diff
@@ -48,119 +46,115 @@ void reverseGeoLookup(degrees_t latitude, degrees_t longitude, taxonomy_data *ta
 	std::unique_ptr<QNetworkReply> reply(rgl.get(request));
 	timer.setSingleShot(true);
 	QObject::connect(&*reply, SIGNAL(finished()), &loop, SLOT(quit()));
-	timer.start(5000);   // 5 secs. timeout
+	timer.start(msTimeout);
 	loop.exec();
 
-	if (timer.isActive()) {
-		timer.stop();
-		if (reply->error() > 0) {
-			report_error("got error accessing geonames.org: %s", qPrintable(reply->errorString()));
-			return;
-		}
-		int v = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
-		if (v < 200 || v >= 300)
-			return;
-		QByteArray fullReply = reply->readAll();
-		QJsonParseError errorObject;
-		QJsonDocument jsonDoc = QJsonDocument::fromJson(fullReply, &errorObject);
-		if (errorObject.error != QJsonParseError::NoError) {
-			report_error("error parsing geonames.org response: %s", qPrintable(errorObject.errorString()));
-			return;
-		}
-		QJsonObject obj = jsonDoc.object();
-		QVariant geoNamesObject = obj.value("geonames").toVariant();
-		QVariantList geoNames = geoNamesObject.toList();
-		if (geoNames.count() > 0) {
-			QVariantMap firstData = geoNames.at(0).toMap();
-			int ri = 0, l3 = -1, lt = -1;
-			if (taxonomy->category == NULL) {
-				taxonomy->category = alloc_taxonomy();
+	if (!timer.isActive()) {
+		report_error("timeout accessing %s", qPrintable(url));
+		QObject::disconnect(&*reply, SIGNAL(finished()), &loop, SLOT(quit()));
+		reply->abort();
+		return QJsonObject{};
+	}
+
+	timer.stop();
+	if (reply->error() > 0) {
+		report_error("got error accessing %s: %s", qPrintable(url), qPrintable(reply->errorString()));
+		return QJsonObject{};
+	}
+	int v = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+	if (v < 200 || v >= 300) {
+		return QJsonObject{};
+	}
+	QByteArray fullReply = reply->readAll();
+	QJsonParseError errorObject;
+	QJsonDocument jsonDoc = QJsonDocument::fromJson(fullReply, &errorObject);
+	if (errorObject.error != QJsonParseError::NoError) {
+		report_error("error parsing JSON response: %s", qPrintable(errorObject.errorString()));
+		return QJsonObject{};
+	}
+	// Success, return JSON response from server
+	return jsonDoc.object();
+}
+
+void reverseGeoLookup(degrees_t latitude, degrees_t longitude, taxonomy_data *taxonomy)
+{
+	const QString geonamesNearbyURL("http://api.geonames.org/findNearbyJSON?lang=%1&lat=%2&lng=%3&radius=50&username=dirkhh");
```

Comment:
Bikeshedding: Since you're refactoring this source file, you could turn those strings into `QStringLiteral`s. I'm fine with leaving them as is.